frontend/ui/widget/inputtext.lua

```diff
@@ -671,13 +672,20 @@ end
 --- Search for a string.
 -- if start_pos not set, starts a search from the next to cursor position
 -- returns first found position or 0 if not found
-function InputText:searchString(str, start_pos)
+function InputText:searchString(str, case_sensitive, start_pos)
     local str_len = string.len(str)
     local char_pos, found = 0, 0
     start_pos = start_pos and (start_pos - 1) or self.charpos
+    if not case_sensitive then
+        str = Utf8Proc.lowercase(str)
+    end
     for i = start_pos, #self.charlist - str_len do
         for j = 1, str_len do
-            if string.lower(self.charlist[i + j]) ~= string.lower(string.sub(str, j, j)) then
+            local char_txt = self.charlist[i + j]
+            if not case_sensitive then
+                char_txt = Utf8Proc.lowercase(char_txt)
+            end
+            if char_txt ~= string.sub(str, j, j) then
```

Comment:
Err, actually, the whole things looks fishy.

str may no longer have the same amount of bytes/codepoints after the lowercasing, so the len is now bogus, and the relationship to charlist is broken.

Reply:
And the second lowercasing is redundant?

Reply:
(No time to investigate, but the TL;DR: is this doesn't look safe).